### ISSUE TITLE
Improved options for executable location and workflow

### ISSUE THREAD

**Issue by Cole-Monnahan-NOAA** [Cole-Monnahan-NOAA/adnuts#35]
Recently @okenk and @jimianelli both mentioned they would like to have better options for a workflow. I'm open to that. Please use this issue to discuss what your preferred workflow is and how it could be implemented in adnuts.

Right now the user specifies the path to a directory that contains all the necessary folders, and the model name. It's assumed the executable is in that folder. Each is then copied to the working directory when running in parallel.

**Comment by Cole-Monnahan-NOAA** [comment]
Yeah for SS3 users this makes a lot of sense. I was working with bespoke models so always made sense that each model would have it's own executable. 

So how about a top level argument like `exe_path`? I'll have to work out the bookkeeping on it throughout the code base, but I think that's a relative easy solution on my end.

**Comment by okenk** [comment]
That would be great.

**Comment by jimianelli** [comment]
exe_path idea great as I use symbolic links from command line to ensure same executable is used for different model configurations (e.g., in linux/osx: ln -s pathtoexecutable/mymod ) in the directory where model is run. 

Additionally, would it make sense to give an option to set the "run_path"? Presently it's set one level up for all the chains in parallel. For example, in my working directory, it might be nice to run the chains in a specific directory within that local directory? Not a big deal but I use a workflow where the working directory I use sets subsidiary folders in there rather than above.

**Comment by Cole-Monnahan-NOAA** [comment]
That's a good suggestion too. I didn't think through very carefully where the chains should be run. Stan uses temporary directories and that's something that might make a lot of sense too. Typically there's no reason to look at them, but I find it useful for debugging when there's an error, and to peak at the adaptation.csv file to see how things are going. But that's probably a very limited case.

So I need to add

1.  A new argument `exe_path` which defaults to `path` but allows the user to locate their executable elsewhere. I think this should also include an option to pull it from the system PATH. Perhaps a value like `exe_path=""` could signify to look for it globally.
2. A new argument `run_path` which dictates where parallel chain folders are created and executed. This would default to R's working directory.  Perhaps a flag like `run_path="temp"` could be used to flag that? This could actually be helpful if you want to run the same model simultaneously, e.g. a RWM and NUTS with 3 chains each. Right now those would conflict.

If anyone, maybe @kellijohnson-noaa, has any other thoughts that would be great. I'll try to get these updates done this summer.
